modules/dns/dns_api.rb

```diff
@@ -47,11 +47,7 @@ class Api < ::Sinatra::Base
 
     delete '/:value/?:type?' do
       name = params[:value]
-      if params[:type]
-        type = params[:type]
-      else
-        type = (name =~ /\.(in-addr|ip6)\.arpa$/) ? "PTR" : "A"
-      end
+      type = params.fetch(:type) { name.end_with?('.in-addr.arpa', '.ip6.arpa') ? 'PTR' : 'A' }
```

Comment:
This fails because `params` does contain the key `type` but has the value `nil`. `fetch` only works if the key isn't present. So this isn't going to work.

A simpler way may be
```suggestion
      type = params[:type]
      type ||= name.end_with?('.in-addr.arpa', '.ip6.arpa') ? 'PTR' : 'A'
```